src/ripple/protocol/STObject.h

```diff
@@ -970,6 +971,27 @@ STObject::operator[](OptionaledField<T> const& of) ->
     return OptionalProxy<T>(this, of.f);
 }
 
+inline
+NetClock::time_point
+getsfCloseTime(STObject const& sle)
```

Comment:
I'm not sure I like this function and its friends below. I can't think of an alternative right this instant, but may be a worth a few minutes of thought to come up with a cleaner interface that reduces the need for duplication.